.github/workflows/periodic-integration-tests.yml

```diff
@@ -8,7 +8,7 @@ on:
 jobs:
   run-k8s-tests:
     runs-on: ubuntu-latest-4-cores
```

Comment:
Can we actually try to downgrade this to ubuntu-latest? I don't think we need anything that powerful if we're just doing concurrency 1. We might need to bump the timeout to something slightly more generous for the first run though.